drivers/led/lp5562.c

```diff
@@ -978,28 +1059,42 @@ static const struct led_driver_api lp5562_led_api = {
-	BUILD_ASSERT(DT_INST_PROP(id, white_output_current) <= LP5562_MAX_CURRENT_SETTING,\
-		"White channel current must be between 0 and 25.5 mA.");	\
-	static const struct lp5562_config lp5562_config_##id = {	\
-		.bus = I2C_DT_SPEC_INST_GET(id),			\
-		.r_current = DT_INST_PROP(id, red_output_current),	\
-		.g_current = DT_INST_PROP(id, green_output_current),	\
-		.b_current = DT_INST_PROP(id, blue_output_current),	\
-		.w_current = DT_INST_PROP(id, white_output_current),	\
-	};								\
-									\
-	struct lp5562_data lp5562_data_##id;				\
-	DEVICE_DT_INST_DEFINE(id, &lp5562_led_init, NULL,		\
-			&lp5562_data_##id,				\
-			&lp5562_config_##id, POST_KERNEL,		\
-			CONFIG_LED_INIT_PRIORITY,			\
-			&lp5562_led_api);				\
+#ifdef CONFIG_PM_DEVICE
+static int lp5562_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		return lp5562_disable(dev);
+	case PM_DEVICE_ACTION_RESUME:
+		return lp5562_enable(dev, false);
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
+#define LP5562_DEFINE(id)                                                                          \
+	BUILD_ASSERT(DT_INST_PROP(id, red_output_current) <= LP5562_MAX_CURRENT_SETTING,           \
+		     "Red channel current must be between 0 and 25.5 mA.");                        \
+	BUILD_ASSERT(DT_INST_PROP(id, green_output_current) <= LP5562_MAX_CURRENT_SETTING,         \
+		     "Green channel current must be between 0 and 25.5 mA.");                      \
+	BUILD_ASSERT(DT_INST_PROP(id, blue_output_current) <= LP5562_MAX_CURRENT_SETTING,          \
+		     "Blue channel current must be between 0 and 25.5 mA.");                       \
+	BUILD_ASSERT(DT_INST_PROP(id, white_output_current) <= LP5562_MAX_CURRENT_SETTING,         \
+		     "White channel current must be between 0 and 25.5 mA.");                      \
+	static const struct lp5562_config lp5562_config_##id = {                                   \
+		.bus = I2C_DT_SPEC_INST_GET(id),                                                   \
+		.r_current = DT_INST_PROP(id, red_output_current),                                 \
+		.g_current = DT_INST_PROP(id, green_output_current),                               \
+		.b_current = DT_INST_PROP(id, blue_output_current),                                \
+		.w_current = DT_INST_PROP(id, white_output_current),                               \
+		.en_pin = GPIO_DT_SPEC_INST_GET_OR(id, en_gpios, {0}),                             \
+	};                                                                                         \
+                                                                                                   \
+	struct lp5562_data lp5562_data_##id;                                                       \
+	DEVICE_DT_INST_DEFINE(id, &lp5562_led_init, NULL, &lp5562_data_##id, &lp5562_config_##id,  \
+			      POST_KERNEL, CONFIG_LED_INIT_PRIORITY, &lp5562_led_api);
```

Comment:
For PM Device to work, we need to add the PM Device instance, then include it in the Device DT Instance. See: https://github.com/zephyrproject-rtos/zephyr/blob/b2475804f8502d21c9d7aeddfd35daa764892fdd/drivers/sensor/lm77/lm77.c#L414-L421